SC-098: Process RFC 8657 CAA Parameters

wthayer opened the pull request:
Update 3.2.2.8 to require that CAs process CAA accounturi and validationmethod parameters defined in RFC 8657

Fixes https://github.com/cabforum/servercert/issues/353

wthayer committed:
Update 3.2.2.8

robstradling commented on docs/BR.md:
```diff
@@ -1082,6 +1082,11 @@ CAs MAY check CAA records at any other time.
+EFFECTIVE DD-MM-YYY:
+When processing CAA records, CAs MUST process the accounturi and validationmethods parameters as specified in RFC 8657. In addition:
+* When the certificate request does not use the ACME protocol defined in RFC 8555, the CA MUST define the recognized format of the accounturi in their CPS.
+* The CA MUST only recognize validationmethods parameters in the format "tlsbr-" + the 3.2.2.4 subsection number that defines the permitted validation method, e.g. "tlsbr-18" represents Agreed-Upon Change to Website v2.
```
Your proposed "tlsbr-" prefixes can only conform to [RFC 8657 section 4](https://www.rfc-editor.org/rfc/rfc8657#name-extensions-to-the-caa-record) if each "tlsbr-" validation method label is registered with the [ACME Validation Methods IANA registry](https://www.iana.org/assignments/acme/acme.xhtml#acme-validation-methods).

However, [RFC 8555 section 9.7.8](https://datatracker.ietf.org/doc/html/rfc8555#section-9.7.8) requires that _"When evaluating a request for an assignment in this registry, the designated expert should ensure that the method being registered has a clear, interoperable definition"_.  The TLS BR DCV methods are "clear" insofar as they specify baseline requirements for DCV, but I wouldn't say that they're "interoperable".  For example, ACME dns-01 and Sectigo's non-ACME DNS-based DCV mechanisms are not interoperable, and yet these are all covered by the TLS BR "3.2.2.4.7 DNS Change" method.  Also, since dns-01 is already present in that IANA registry, ISTM that the _"does not overlap with existing validation methods"_ requirement would be a roadblock to registering a "tlsbr-7" label.

Furthermore, requiring that only "tlsbr-" labels be recognised by the CA would mean that the primary ACME use cases described by RFC 8657 (e.g., "validationmethods=dns-01") would no longer be permitted!

Note that to avoid this IANA registry requirement, [RFC 8657 section 4](https://www.rfc-editor.org/rfc/rfc8657#name-extensions-to-the-caa-record) requires that _'the CA MUST use labels beginning with the string "ca-", which are defined to have CA-specific meaning.'_  I think "CA-specific" is entirely appropriate for the use case we're discussing in this PR, because each CA takes the baseline requirements for DCV (described in the 3.2.2.4 subsections) and creates their own specific implementation that is interoperable (between the CA and their Subscribers).

I propose changing this line as follows:
```suggestion
* The CA MUST define each recognized validationmethods label, along with the corresponding 3.2.2.4 subsection number, in their CPS.
```

wthayer replied:
Agreed that whatever we do has to begin with 'ca-' to comply with 8657. Your proposal simplifies this requirement at the expense of making the semantics CA-specific, potentially creating confusion and hindering adoption. Since I have yet to devise a solution to the dns-01 vs ca-tlsbr-7 confusion, I will propose your approach. Thank you!

wthayer replied:
We discussed this on today's Validation SC call and consensus was that we should define the format for the validationmethods label, e.g. 'ca-tlsbr-<method #>' One argument for this was that a standardized format would be more likely to result in automation support.

In addition, @CBonnell suggested that we define the URI format for the accounturi parameter.

I plan to draft changes to this PR to address this feedback.

wthayer commented on docs/BR.md:
```diff
@@ -1082,6 +1082,11 @@ CAs MAY check CAA records at any other time.
 
+EFFECTIVE DD-MM-YYY:
+When processing CAA records, CAs MUST process the accounturi and validationmethods parameters as specified in RFC 8657. In addition:
+* When the certificate request does not use the ACME protocol defined in RFC 8555, the CA MUST define the recognized format of the accounturi in their CPS.
```
Agreed, your wording is better!

wthayer committed:
Update CPS requirement

Co-authored-by: Rob Stradling <rob@sectigo.com>

wthayer committed:
Move to ca-defined method labels

Co-authored-by: Rob Stradling <rob@sectigo.com>

wthayer committed:
Add 8657 compliance

- validationmethod labels must comply with section 4 of RFC 8657
- Update effective date format
- Add 'this section' to CPS requirements.

wthayer committed:
Add 8555 and 8657 references

wthayer committed:
Link section refs

geegeea commented on docs/BR.md:
```diff
@@ -1082,6 +1086,10 @@ CAs MAY check CAA records at any other time.
 
 When processing CAA records, CAs MUST process the issue, issuewild, and iodef property tags as specified in RFC 8659, although they are not required to act on the contents of the iodef property tag. Additional property tags MAY be supported, but MUST NOT conflict with or supersede the mandatory property tags set out in this document. CAs MUST respect the critical flag and not issue a certificate  ...
 
+*Effective September 15, 2025*, when processing CAA records, CAs MUST process the accounturi and validationmethods parameters as specified in RFC 8657. In addition:
```
RFC 8657 section 5.6 requires DNSSEC to be validated while performing CAA checks; under the current text, CAs would need to start validating DNSSEC for all CAA DNS queries; is this intentional?

We have seen that in practice many domains that are able to validate under current rules would begin to fail with strict DNSSEC enforcement. GTS would love to see DNSSEC used more widely and configured properly more often, but more time is likely needed before strict DNSSEC checks cause minimal issues for legitimate requesters.

wthayer replied:
Thank you, this is an excellent point. We discussed this today on the Validation SC call and decided that the cleanest solution is to merge this ballot with @clintwilson's proposal to require DNSSEC as drafted at https://github.com/cabforum/servercert/compare/main..Require-DNSSEC-for-CAA-Lookups

wthayer replied:
Clint's DNSSEC ballot has been incorporated.

wthayer committed:
Incorporate Clint's DNSSEC requirements

wthayer committed:
Specify validationmethods format

wthayer committed:
Ben's 3.2.2.8 wording suggestions

wthayer committed:
Reformat 3.2.2.8 and add subsections

wthayer committed:
Attempt to clarify dns-01 vs ca-dv-7 processing.

wthayer commented:
Updated based on 24-Jan Validation meeting:
- still specifying the CA-specific label format. consensus was that this does not violate the RFC
- adopted Ben's wording
- rearranged 3.2.2.8 and added subsections
- Changed MUST date to 2027 for parameters. Left the 2026 date for DNSSEC since it's arguably a clarification
- Drafted a recommendation that CAs accept validationmethods labels from ACME or the BRs

wthayer committed:
Incorporate today's discussion

wthayer committed:
dv to tbr in example

wthayer committed:
Align CPS section requirement

wthayer committed:
Remove section link

wthayer committed:
Allow multiple accounturi formats

Co-authored-by: Corey Bonnell <corey.j.bonnell@outlook.com>

wthayer committed:
Add effective date for CPS + label format

wthayer committed:
Move 3.2.2.8 to 4.2.1.1

wthayer committed:
Update 3.2.2.9 reference

wthayer committed:
Revert DNSSEC requirements

wthayer committed:
Fix heading levels

wthayer committed:
Move to 4.2.2 like SBRs

aarongable commented on docs/BR.md:
```diff
@@ -1236,6 +1219,43 @@ If a Delegated Third Party fulfills any of the CA's obligations under this secti
+
+In addition, *Effective March 15, 2026*, if the CA processes the accounturi and validationmethods parameters:
+* If the CA accepts certificate requests via any protocol other than the ACME protocol defined in RFC 8555, the CA MUST define the supported format(s) of the accounturi in Section 4.2 of their CP and/or CPS.
+* If the CA accepts certificate requests via any protocol other than the ACME protocol defined in RFC 8555, the CA MUST interpret and process validationmethods labels formed by concatenating the string ‘ca-tbr-’ with the BR 3.2.2.4 subsection number, e.g. ‘ca-tbr-7’ represents the DNS method described in TLS BR 3.2.2.4.7. If a CA performs domain validation using a mechanism that can be represente ...
```
Echoing my comment from the call this morning: I think this sentence comes a little close to the line of "well, CA Foo doesn't _quite_ implement RFC 8555 (e.g. they don't provide a list of all orders for an account), so they qualify here and need to support these validationmethods labels". I think it's clear that this is not the _intent_ of this sentence, so I'm wondering if there's any minor change we can make to make this clear.

wthayer committed:
Remove strict RFC 8555 compliance requirement.

wthayer committed:
Add To-do for 3.2.2.8.1

orangepizza commented on docs/BR.md:
```diff
@@ -1236,6 +1219,45 @@ If a Delegated Third Party fulfills any of the CA's obligations under this secti
+
+In addition, *Effective March 15, 2026*, if the CA processes the accounturi and validationmethods parameters:
+* If the CA accepts certificate requests via any protocol other than the ACME protocol, the CA MUST define the supported format(s) of the accounturi in Section 4.2 of their CP and/or CPS.
+* If the CA accepts certificate requests via any protocol other than the ACME protocol, the CA MUST interpret and process validationmethods labels formed by concatenating the string ‘ca-tbr-’ with the BR 3.2.2.4 subsection number, e.g. ‘ca-tbr-7’ represents the DNS method described in TLS BR 3.2.2.4.7. If a CA performs domain validation using a mechanism that can be represented by multiple labels ...
```
1. Can 'http-01' considered to represent 3.2.2.4.18? while it'd strictly mean 3.2.2.4.19, it's reasonable to assume user meant any http update challenges.
2. Can ACME running CA use 'ca-tbr-19' as permission to use http-01 challenge? this paragraph doesn't apply if a CA doesn't offer something else, isn't it?

aarongable replied:
> 1. Can 'http-01' considered to represent 3.2.2.4.18? while it'd strictly mean 3.2.2.4.19, it's reasonable to assume user meant any http update challenges.

No, 3.2.2.4.18 and 3.2.2.4.19 are in fact different validation methods (the expect the token to be in a different location and to look different) so permission to use one method cannot be interpreted as permission to use the other.

>     2. Can ACME running CA use 'ca-tbr-19' as permission to use http-01 challenge? this paragraph doesn't apply if a CA doesn't offer something else, isn't it?

Yes, this paragraph simply says that non-ACME CAs MUST respect those designations; therefore an ACME CA MAY also respect them, as long as it appropriately documents that fact in its CPS.

wthayer committed:
Merge branch 'main' into SC-XX-Process-RFC-8657-CAA-Parameters

wthayer committed:
Merge in SC-085 changes

wthayer committed:
Merge Grace's CAA Parameters proposal

wthayer committed:
Remove duplicated MPIC language

wthayer committed:
Update effective dates

wthayer committed:
Incorporate 3.2.2.9 change

wthayer committed:
Update base (#653)

Co-authored-by: Ryan Dickson <ryandickson@google.com>
Co-authored-by: Dimitris Zacharopoulos <dzacharo@users.noreply.github.com>
Co-authored-by: Martijn Katerbarg <martijn.katerbarg@sectigo.com>

wthayer committed:
Align with Cleanup ballot

wthayer committed:
Merge branch 'main' into SC-XX-Process-RFC-8657-CAA-Parameters

wthayer committed:
Add 'acct' URI scheme reference

wthayer committed:
Merge branch 'main' into SC-XX-Process-RFC-8657-CAA-Parameters

wthayer committed:
SC95 formatting changes

wthayer committed:
Sync 3.2.2.9 with SC95

wthayer committed:
sync with sc95

wthayer committed:
change example

wthayer committed:
Address label case sensitivity

wthayer committed:
More case sensitivity language

wthayer committed:
Remove past DNSSEC effective dates

wthayer committed:
Draft ACME accounturi override

ChristopherRC commented on docs/BR.md:
```diff
@@ -1367,6 +1337,61 @@ CAs SHALL NOT issue Certificates containing Internal Names or Reserved IP Addres
+
+In addition, *Effective 2027-03-15*:
+- If the CA accepts certificate requests via any protocol other than the ACME protocol defined in [RFC 8555](https://datatracker.ietf.org/doc/html/rfc8555), the CA MUST define the supported format of the `accounturi` in Section 4.2 of their CP and/or CPS, and SHOULD comply with the 'acct' URI scheme defined in [RFC 7565](https://datatracker.ietf.org/doc/html/rfc7565).
+- For certificate requests made using the ACME protocol, the CA MAY also define an 'accounturi' format following the requirements in the prior bullet. If the CA accepts a self-defined 'accounturi' format for ACME requests, the CA MUST maintain a mapping between the self-defined label and the ACME accounts for which the self-defined label authorizes issuance. Note: this provision overrides RFC 865 ...
```
What about something like this?

```suggestion
- For certificate requests made using the ACME protocol, the CA MAY permit the 'accounturi' parameter to identify a primary organizational account (the "Parent Account"). As an explicit exception to Section 3 of RFC 8657, the CA MAY issue a certificate requested by a different account (the "Subordinate ACME Account") if and only if the CA ensures all of the following:
1. The CA maintains an internal, auditable mapping that binds the Subordinate ACME Account to the Parent Account identified by the 'accounturi'.
2. The CA has cryptographically or administratively verified that the Parent Account explicitly authorized the Subordinate ACME Account to obtain certificates under this mapping.
3. The CA retains audit logs demonstrating this authorization and mapping for the standard data retention period required by these Requirements.
```

The intent being:

- Explicitly defining the relationship as a "Parent Account" and a "Subordinate ACME Account" in an attempt to remove ambiguity about what a "self-defined label" is.
- Point 2 attempts to ensure that a CA's automated system cannot arbitrarily map two accounts together. It requires the CA to prove that the Parent Account owner actually authorized the creation of the child account.
- Point 3 attempts to ensure that an auditor reviewing a certificate where the requested ACME account doesn't strictly match the DNS accounturi can ask the CA for the logs to prove the issuance was valid.
- Removing the "Note:" and referencing the specific exception to Section 3 of RFC 8657 makes the deviation more of a normative requirement.

sheurich replied:
+1 on the wording change from "this provision overrides RFC 8657" to an explicit exception to Section 3. The current "overrides" contradicts the "also" earlier in the same bullet and leaves room to read the text as permission to stop accepting the ACME account URL, which I don't believe is the intent. "Explicit exception to Section 3 of RFC 8657" scopes the deviation precisely.

(Narrow endorsement of the framing only — no opinion on the Parent/Subordinate scaffolding.)

geegeea commented on docs/BR.md:
```diff
@@ -1367,6 +1337,61 @@ CAs SHALL NOT issue Certificates containing Internal Names or Reserved IP Addres
+*Effective 2027-03-15*, when processing CAA records, CAs MUST process the `accounturi` and `validationmethod`s parameters as specified in [RFC 8657](https://datatracker.ietf.org/doc/html/rfc8657).
+
+In addition, *Effective 2027-03-15*:
+- If the CA accepts certificate requests via any protocol other than the ACME protocol defined in [RFC 8555](https://datatracker.ietf.org/doc/html/rfc8555), the CA MUST define the supported format of the `accounturi` in Section 4.2 of their CP and/or CPS, and SHOULD comply with the 'acct' URI scheme defined in [RFC 7565](https://datatracker.ietf.org/doc/html/rfc7565).
```
```suggestion
- If the CA accepts certificate requests via any protocol other than a protocol functionally equivalent to the ACME protocol defined in [RFC 8555](https://datatracker.ietf.org/doc/html/rfc8555), the CA MUST define the supported format of the `accounturi` in Section 4.2 of their CP and/or CPS, and SHOULD comply with the 'acct' URI scheme defined in [RFC 7565](https://datatracker.ietf.org/doc/html/rfc7565).
```

Echoing @aarongable 's older [comment](https://github.com/cabforum/servercert/pull/567#discussion_r1993768921), the suggested change is intended to provide necessary flexibility for CAs whose implementations are based on the ACME protocol but may contain minor deviations from the strict requirements of RFC 8555. For example, a CA may not accept the optional notBefore and notAfter fields of a newOrder request payload. That shouldn't disqualify them from being considered an ACME CA. Similar suggestion on line 1375 below.

XolphinMartijn replied:
@geegeea  I get the thought here, but "functionally equivalent to" is very much open to interpretation. Is a regular REST API functionally equivalent to ACME? Cause the end result can be the same..

Also a minor nit, but:
```suggestion
- If the CA accepts certificate requests via any method other than the ACME protocol defined in [RFC 8555](https://datatracker.ietf.org/doc/html/rfc8555), the CA MUST define the supported format of the `accounturi` in Section 4.2 of their CP and/or CPS, and SHOULD comply with the 'acct' URI scheme defined in [RFC 7565](https://datatracker.ietf.org/doc/html/rfc7565).
```

I would suggest the above option, rather than "protocol"

geegeea replied:
I see your point, but the current wording is ambiguous for CAs with minor deviations from the spec. As an example, see the list of Boulder's divergences from ACME: https://github.com/letsencrypt/boulder/blob/main/docs/acme-divergences.md. I would consider Boulder very much an ACME server but the above wording makes it unclear by implying strict RFC 8555 compliance. Since the core of this requirement is how the account is identified, would it be better to say: 'If the CA does not identify the Subscriber account via an ACME Account URL as described in RFC 8555'? This focuses on the URI format rather than full RFC 8555 compliance.
```suggestion
- If the CA does not identify the Subscriber account via an ACME Account URL as described in [RFC 8555](https://datatracker.ietf.org/doc/html/rfc8555), the CA MUST define the supported format of the `accounturi` in Section 4.2 of their CP and/or CPS, and SHOULD comply with the 'acct' URI scheme defined in [RFC 7565](https://datatracker.ietf.org/doc/html/rfc7565).
```

wthayer replied:
Accepted Gurleen's second suggestion. Thank you!

geegeea commented on docs/BR.md:
```diff
@@ -1367,6 +1337,61 @@ CAs SHALL NOT issue Certificates containing Internal Names or Reserved IP Addres
+In addition, *Effective 2027-03-15*:
+- If the CA accepts certificate requests via any protocol other than the ACME protocol defined in [RFC 8555](https://datatracker.ietf.org/doc/html/rfc8555), the CA MUST define the supported format of the `accounturi` in Section 4.2 of their CP and/or CPS, and SHOULD comply with the 'acct' URI scheme defined in [RFC 7565](https://datatracker.ietf.org/doc/html/rfc7565).
+- For certificate requests made using the ACME protocol, the CA MAY also define an 'accounturi' format following the requirements in the prior bullet. If the CA accepts a self-defined 'accounturi' format for ACME requests, the CA MUST maintain a mapping between the self-defined label and the ACME accounts for which the self-defined label authorizes issuance. Note: this provision overrides RFC 865 ...
+- If the CA accepts certificate requests via any protocol other than the ACME protocol, the CA MUST interpret and process `validationmethods` labels formed by concatenating the string ‘ca-tbr-’ with the BR 3.2.2.4 subsection number, e.g. ‘ca-tbr-7’ represents the DNS method described in TLS BR 3.2.2.4.7. If a CA performs domain validation using a mechanism that can be represented by multiple labe ...
```
```suggestion
- If the CA supports domain validation methods that are not registered in the [IANA ACME Validation Methods registry](https://www.iana.org/assignments/acme/acme.xhtml#acme-validation-methods), the CA MUST interpret and process `validationmethods` labels formed by concatenating the string ‘ca-tbr-’ with the BR 3.2.2.4 subsection number, e.g. ‘ca-tbr-7’ represents the DNS method described in TLS BR 3.2.2.4.7. If a CA performs domain validation using a mechanism that can be represented by multiple labels (e.g. 'http-01' and 'ca-tbr-19'), the CA SHOULD accept any of the labels as granting permission to issue.
```

Same suggestion as above on line 1373 to avoid a situation where a functionally equivalent ACME-based service is unintentionally forced into the 'any protocol other than the ACME protocol' category.
Edit: Updated suggestion to address @XolphinMartijn 's comment above.

sheurich commented on docs/BR.md:
```diff
@@ -1151,56 +1157,20 @@ Databases maintained by the CA, its owner, or its affiliated companies do not qu
 
-The set of responses from the relied upon Network Perspectives MUST provide the CA with the necessary information to allow it to affirmatively assess:
-
 a. the presence of the expected 1) Random Value, 2) Request Token, 3) IP Address, 4) Contact Address, or 5) Persistent DCV TXT Record, as required by the relied upon validation method specified in [Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control) and [Section 3.2.2.5](#3225-authentication-for-an-ip-address); and
```
The reorganization drops the sentence introducing items `a.` and `b.`:

```diff
- The set of responses from the relied upon Network Perspectives MUST provide the CA with the necessary information to allow it to affirmatively assess:
-
  a. the presence of the expected 1) Random Value, ...
  b. the CA's authority to issue to the requested domain(s), ...
```

Without the preamble, items `a.` and `b.` are grammatically incomplete fragments, and the normative MUST governing MPIC assessment is lost.

Suggested fix — restore the preamble above item `a.`:

```suggestion
The set of responses from the relied upon Network Perspectives MUST provide the CA with the necessary information to allow it to affirmatively assess:

a. the presence of the expected 1) Random Value, 2) Request Token, 3) IP Address, 4) Contact Address, or 5) Persistent DCV TXT Record, as required by the relied upon validation method specified in [Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control) and [Section 3.2.2.5](#3225-authentication-for-an-ip-address); and
```

sheurich commented on docs/BR.md:
```diff
@@ -1151,56 +1157,20 @@ Databases maintained by the CA, its owner, or its affiliated companies do not qu
-
 a. the presence of the expected 1) Random Value, 2) Request Token, 3) IP Address, 4) Contact Address, or 5) Persistent DCV TXT Record, as required by the relied upon validation method specified in [Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control) and [Section 3.2.2.5](#3225-authentication-for-an-ip-address); and
-b. the CA's authority to issue to the requested domain(s), as specified in [Section 3.2.2.8](#3228-caa-records).
+b. the CA's authority to issue to the requested domain(s), as specified in [Section 4.2.2.1](#4221-caa-records).
```
The anchor target `#4221-caa-records` doesn't match the generated slug for the new heading (`#### 4.2.2.1 CAA record processing` → `#4221-caa-record-processing`). The §3.2.2.8 stub above gets this right; this link does not.

```suggestion
b. the CA's authority to issue to the requested domain(s), as specified in [Section 4.2.2.1](#4221-caa-record-processing).
```

sheurich commented on docs/BR.md:
```diff
@@ -1367,6 +1337,61 @@ CAs SHALL NOT issue Certificates containing Internal Names or Reserved IP Addres
+###### 4.2.2.1.3 CAA Parameters
+
+When processing CAA records, CAs SHOULD process the `accounturi` and `validationmethods` parameters as specified in [RFC 8657](https://datatracker.ietf.org/doc/html/rfc8657).
+*Effective 2027-03-15*, when processing CAA records, CAs MUST process the `accounturi` and `validationmethod`s parameters as specified in [RFC 8657](https://datatracker.ietf.org/doc/html/rfc8657).
```
Backtick misplaced — the trailing `s` falls outside the code span, so this renders as `validationmethod`s in body font. The SHOULD line above (line 1369) has it right.

```suggestion
*Effective 2027-03-15*, when processing CAA records, CAs MUST process the `accounturi` and `validationmethods` parameters as specified in [RFC 8657](https://datatracker.ietf.org/doc/html/rfc8657).
```

sheurich commented on docs/BR.md:
```diff
@@ -1367,6 +1337,61 @@ CAs SHALL NOT issue Certificates containing Internal Names or Reserved IP Addres
+
+###### 4.2.2.1.2 CAA Multi-Perspective Issuance Corroboration
+
+Some methods relied upon for validating the Applicant's ownership or control of the subject domain(s) (see [Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control)) or IP address(es) (see [Section 3.2.2.5](#3225-authentication-for-an-ip-address)) to be listed in a certificate require CAA records to be retrieved and processed from additional remote Network Perspectives before Certifi ...
```
The phrase "as defined in this section" used to unambiguously refer to §3.2.2.8, which contained both this paragraph and the acceptable-lookup-failure definition. After the split, this paragraph lives in §4.2.2.1.2 but the definition is in the parent §4.2.2.1 — an explicit cross-reference would remove the ambiguity.

```suggestion
Some methods relied upon for validating the Applicant's ownership or control of the subject domain(s) (see [Section 3.2.2.4](#3224-validation-of-domain-authorization-or-control)) or IP address(es) (see [Section 3.2.2.5](#3225-authentication-for-an-ip-address)) to be listed in a certificate require CAA records to be retrieved and processed from additional remote Network Perspectives before Certificate issuance (see [Section 3.2.2.9](#3229-multi-perspective-issuance-corroboration)). To corroborate the Primary Network Perspective, a remote Network Perspective's CAA check response MUST be interpreted as permission to issue, regardless of whether the responses from both Perspectives are byte-for-byte identical. Additionally, a CA MAY consider the response from a remote Network Perspective as corroborating if one or both of the Perspectives experience an acceptable CAA record lookup failure, as defined in [Section 4.2.2.1](#4221-caa-record-processing).
```

sheurich commented on docs/BR.md:
```diff
@@ -1367,6 +1337,61 @@ CAs SHALL NOT issue Certificates containing Internal Names or Reserved IP Addres
+
+CAs MUST document potential issuances that were prevented by a CAA record in sufficient detail to provide feedback to the CA/Browser Forum on the circumstances, and SHOULD dispatch reports of such issuance requests to the contact(s) stipulated in the CAA iodef record(s), if present. CAs are not expected to support URL schemes in the iodef record other than mailto: or https:.
+
+###### 4.2.2.1.2 CAA Multi-Perspective Issuance Corroboration
```
All three subsections (`4.2.2.1.2`, `4.2.2.1.3`, `4.2.2.1.4`) use `######` (h6), skipping h5. The existing BR convention at this depth is h5 — see `##### 3.2.2.8.1 DNSSEC Validation of CAA Records`. Applying the same fix to lines 1367 and 1378 will restore consistency with the rest of §3.2.2.x.

```suggestion
##### 4.2.2.1.2 CAA Multi-Perspective Issuance Corroboration
```

wthayer committed:
For ACME, add accounturi parent/child option

Co-authored-by: Chris Clements <cclements@google.com>

wthayer committed:
Remove implicit requirement to exactly implement RFC 8555

Co-authored-by: Gurleen Grewal <gurleen.grewal@gmail.com>

wthayer committed:
Remove implicit requirement to exactly implement RFC 8555

Co-authored-by: Gurleen Grewal <gurleen.grewal@gmail.com>

wthayer committed:
Replace missing sentence.

Co-authored-by: Shiloh Heurich <1778483+sheurich@users.noreply.github.com>

wthayer committed:
Fix links

Co-authored-by: Shiloh Heurich <1778483+sheurich@users.noreply.github.com>

wthayer committed:
Fix formatting

Co-authored-by: Shiloh Heurich <1778483+sheurich@users.noreply.github.com>

wthayer committed:
Clarify reference

Co-authored-by: Shiloh Heurich <1778483+sheurich@users.noreply.github.com>

wthayer committed:
Fix formatting

Co-authored-by: Shiloh Heurich <1778483+sheurich@users.noreply.github.com>

wthayer committed:
Fix formatting

wthayer committed:
Fix section numbering

dzacharo committed:
Merge branch 'main' into SC-XX-Process-RFC-8657-CAA-Parameters

dzacharo committed:
Merge branch 'main' into SC-XX-Process-RFC-8657-CAA-Parameters

dzacharo committed:
Update version number and tables in sections 1.2.1, 1.2.2
